tests/test_repository_lib.py

```diff
@@ -463,37 +465,136 @@ def test_generate_snapshot_metadata(self):
+
+
+
+  def test_generate_snapshot_metadata_without_hashes(self):
+    metadata_directory, version, expiration_date, storage_backend = \
+      self._setup_generate_snapshot_metadata_test()
+
+    snapshot_metadata = \
+      repo_lib.generate_snapshot_metadata(metadata_directory, version,
+                                          expiration_date,
+                                          storage_backend,
+                                          consistent_snapshot=False,
+                                          use_hashes=False)
+    self.assertTrue(tuf.formats.SNAPSHOT_SCHEMA.matches(snapshot_metadata))
+
+    metadata_files_info_dict = snapshot_metadata['meta']
+    for metadata_filename in sorted(os.listdir(metadata_directory), reverse=True):
+
+      # In the metadata_directory, there are files with format:
+      # 1.root.json. The prefix number should be removed.
+      striped_filename, version = \
+        repo_lib._strip_version_number(metadata_filename,
+                                      consistent_snapshot=True)
+
+      # In the repository, the file "role_file.xml" have been added to make
+      # sure that non-json files aren't loaded. This file should be filtered.
+      if striped_filename.endswith('.json'):
+        if striped_filename not in TOP_LEVEL_METADATA_FILES:
+
+          # Check that hashes is not calculated but length is
+          self.assertIn('length', metadata_files_info_dict[striped_filename])
+          self.assertNotIn('hashes', metadata_files_info_dict[striped_filename])
+
+
+
+  def test_generate_snapshot_metadata_without_hashes_and_length(self):
+    metadata_directory, version, expiration_date, storage_backend = \
+      self._setup_generate_snapshot_metadata_test()
+
+    snapshot_metadata = \
+      repo_lib.generate_snapshot_metadata(metadata_directory, version,
+                                          expiration_date,
+                                          storage_backend,
+                                          consistent_snapshot=False,
+                                          use_length=False,
+                                          use_hashes=False)
+    self.assertTrue(tuf.formats.SNAPSHOT_SCHEMA.matches(snapshot_metadata))
+
+    metadata_files_info_dict = snapshot_metadata['meta']
+    for metadata_filename in sorted(os.listdir(metadata_directory), reverse=True):
+
+      # In the metadata_directory, there are files with format:
+      # 1.root.json. The prefix number should be removed.
+      striped_filename, version = \
+        repo_lib._strip_version_number(metadata_filename,
+                                      consistent_snapshot=True)
+
+      # In the repository, the file "role_file.xml" have been added to make
+      # sure that non-json files aren't loaded. This file should be filtered.
+      if striped_filename.endswith('.json'):
+        if striped_filename not in TOP_LEVEL_METADATA_FILES:
+
```

Comment:
Nit: you're inconsistent in your usage of a blank line after an if statement. Personally, I prefer no blank line. But here with no blank line following the first `if` and a blank line following the second is jarring.

Reply:
Yes in this situation it's awkward. I will remove the blank line.
The reason I added it so that the assertions could be easier to read, but I understand your confusion.